src/main/java/teammates/common/datatransfer/attributes/FeedbackQuestionAttributes.java

```diff
@@ -633,6 +603,26 @@ public String getQuestionAdditionalInfoHtml() {
         return getQuestionDetails().getQuestionAdditionalInfoHtml(questionNumber, "");
     }
 
+    private static FeedbackQuestionDetails deserializeFeedbackQuestionDetails(String questionDetailsInJson,
+                                                                              FeedbackQuestionType questionType) {
+        if (questionType == FeedbackQuestionType.TEXT) {
```

Comment:
Why we don't stick to the old way?

```java
// For old Text questions, the questionText simply contains the question, not a JSON	        return questionDetails.getDeepCopy();
        if (questionType == FeedbackQuestionType.TEXT && !isValidJsonString(questionMetaData)) {	
            return new FeedbackTextQuestionDetails(questionMetaData);	
        }	
        return JsonUtils.fromJson(questionMetaData, getFeedbackQuestionDetailsClass());
```

Reply:
During my development, I found that if the string is a single word, e.g "question", it is still a valid Json String. The new method can make use of the `FeedbackTextQuestionDetails.class` to get a stronger validation

Reply:
> I found that if the string is a single word, e.g "question", it is still a valid Json String.

This sounds like a problem of `isValidJsonString` to me.

> The new method can make use of the `FeedbackTextQuestionDetails.class` to get a stronger validation

What if `questionDetailsInJson ` is not `FeedbackTextQuestionDetails.class`

Reply:
The logic is: only if the question type is text, it will move to `deserializeTextQuestion()`. Other types will finish the work inside `deserializeQuestion()`

Reply:
Does the `deserializeFeedbackTextQuestionDetails ` solve the problem you described?

Reply:
Yes, and I also add tests to illustrate it